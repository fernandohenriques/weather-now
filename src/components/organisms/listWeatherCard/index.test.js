import React from 'react';
import { shallow } from 'enzyme';
import ListWeatherCard from './index';

describe('<ListWeatherCard />', () => {
  const citiesMock = [
    { id: 1, name: 'Nuuk', country: 'GL', temperature: -5, updatedAt: '02:48:32 PM' },
    { id: 2, name: 'Ubirici', country: 'BR', temperature: 19, humidity: 75, pressure: 892, updatedAt: '02:48:32 PM' },
  ];

  it('Snapshot testing', () => {
    const wrapper = shallow(<ListWeatherCard cities={citiesMock} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render WeatherCard component', () => {
    const wrapper = shallow(<ListWeatherCard cities={citiesMock} />);

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('WeatherCard')).toHaveLength(2);
  });
});
