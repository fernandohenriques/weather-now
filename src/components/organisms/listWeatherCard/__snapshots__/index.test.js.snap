// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListWeatherCard /> Snapshot testing 1`] = `
<section
  className="container"
>
  <WeatherCard
    key="1"
    temperature={-5}
    title="Nuuk, GL"
    updatedAt="02:48:32 PM"
  />
  <WeatherCard
    humidity={75}
    key="2"
    pressure={892}
    temperature={19}
    title="Ubirici, BR"
    updatedAt="02:48:32 PM"
  />
</section>
`;
